src/RealFourierArray.jl

```diff
@@ -49,8 +49,11 @@ end
 check_real(A::RealFourierArray{Treal, T}) where {Treal, T <: Real} = nothing
 check_real(A::RealFourierArray) = check_real(A.real)
 
-# lie blatantly to the user, as recommended by the docs
-Base.fieldnames(RealFourierArray) = (:basis, :real, :fourier)
+function Base.propertynames(array::RealFourierArray, private=false)
+    ret = [:basis, :real, :fourier]
+    private && append!(ret, fieldnames(RealFourierArray))
```

Comment:
boo mutation :-p
`private ? append(ret, fieldnames(RealFourierArray)) : ret` ?

Reply:
There is no `append` ... so if you don't want to repeat ret I don't see a way without mutation.

Reply:
Really, that's weird. `[ret..., fieldnames(RealFourierArray)...]`

Reply:
Really, what's the point? This function is not performance-critical and what you propose either allocates twice as well (if ret is an array) or is not type-stable (once returns array, once tuple if ret is a tuple). Also it's harder to read. I'm not convinced.

Reply:
Ret should not be a tuple? But yeah I'm fine with what you wrote, it's just my part-time crusade against mutation.

Reply:
No. either tuple or vector. Both is allowed explicitly.